ts/state/selectors/conversations.ts

```diff
@@ -140,90 +139,32 @@ export const getSortedMessagesTypesOfSelectedConversation = createSelector(
           ? messageTimestamp
           : undefined;
 
-      const common = { showUnreadIndicator: isFirstUnread, showDateBreak };
-
-      if (msg.propsForDataExtractionNotification) {
-        return {
-          ...common,
-          message: {
-            messageType: 'data-extraction',
-            props: { ...msg.propsForDataExtractionNotification, messageId: msg.propsForMessage.id },
-          },
-        };
-      }
-
-      if (msg.propsForMessageRequestResponse) {
-        return {
-          ...common,
-          message: {
-            messageType: 'message-request-response',
-            props: { ...msg.propsForMessageRequestResponse, messageId: msg.propsForMessage.id },
-          },
-        };
-      }
-
-      if (msg.propsForGroupInvitation) {
-        return {
-          ...common,
-          message: {
-            messageType: 'group-invitation',
-            props: { ...msg.propsForGroupInvitation, messageId: msg.propsForMessage.id },
-          },
-        };
-      }
-
-      if (msg.propsForGroupUpdateMessage) {
-        return {
-          ...common,
-          message: {
-            messageType: 'group-notification',
-            props: { ...msg.propsForGroupUpdateMessage, messageId: msg.propsForMessage.id },
-          },
-        };
-      }
-
-      if (msg.propsForTimerNotification) {
-        return {
-          ...common,
-          message: {
-            messageType: 'timer-notification',
-            props: { ...msg.propsForTimerNotification, messageId: msg.propsForMessage.id },
-          },
-        };
-      }
-
-      if (msg.propsForCallNotification) {
-        return {
-          ...common,
-          message: {
-            messageType: 'call-notification',
-            props: {
-              ...msg.propsForCallNotification,
-              messageId: msg.propsForMessage.id,
-            },
-          },
-        };
-      }
+      const common = {
+        showUnreadIndicator: isFirstUnread,
+        showDateBreak,
+        messageId: msg.propsForMessage.id,
+      };
 
-      if (msg.propsForInteractionNotification) {
-        return {
-          ...common,
-          message: {
-            messageType: 'interaction-notification',
-            props: {
-              ...msg.propsForInteractionNotification,
-              messageId: msg.propsForMessage.id,
-            },
-          },
-        };
-      }
+      const messageType = msg.propsForDataExtractionNotification
```

Comment:
brrrrrrrruuuuuuuuuuhhhhhhhhhhhh

Reply:
~~Do you think you can do something similar to this ?~~

e.g.

```tsx
  const nodeComps: Record<number, (props: NodeGraphProps) => React.JSX.Element> = {
    1: NodeGraph1,
    2: NodeGraph2,
    3: NodeGraph3,
    4: NodeGraph4,
    5: NodeGraph5,
    6: NodeGraph6,
    7: NodeGraph7,
    8: NodeGraph8,
    9: NodeGraph9,
    10: NodeGraph10,
  };

  const NodeComp = nodeComps[count];
  
  // ...

return(<NodeComp />);
```

Reply:
Actually I dont think the above is possible. You might have the best solution if you need to actually check if the props are defined 😅